src/jaxsim/api/data.py

```diff
@@ -133,56 +132,73 @@ def build(
                 The base angular velocity in the selected representation.
             joint_velocities: The joint velocities.
             standard_gravity: The standard gravity constant.
-            contact: The state of the soft contacts.
             contacts_params: The parameters of the soft contacts.
             velocity_representation: The velocity representation to use.
             time: The time at which the state is created.
+            extended_ode_state:
+                Additional user-defined state variables that are not part of the
+                standard `ODEState` object. Useful to extend the system dynamics
+                considered by default in JaxSim.
 
         Returns:
-            A `JaxSimModelData` object with the given state.
+            A `JaxSimModelData` initialized with the given state.
         """
 
         base_position = jnp.array(
-            base_position if base_position is not None else jnp.zeros(3)
+            base_position if base_position is not None else jnp.zeros(3),
+            dtype=float,
         ).squeeze()
 
         base_quaternion = jnp.array(
-            base_quaternion
-            if base_quaternion is not None
-            else jnp.array([1.0, 0, 0, 0])
+            (
+                base_quaternion
+                if base_quaternion is not None
+                else jnp.array([1.0, 0, 0, 0])
+            ),
+            dtype=float,
         ).squeeze()
 
         base_linear_velocity = jnp.array(
-            base_linear_velocity if base_linear_velocity is not None else jnp.zeros(3)
+            base_linear_velocity if base_linear_velocity is not None else jnp.zeros(3),
+            dtype=float,
         ).squeeze()
 
         base_angular_velocity = jnp.array(
-            base_angular_velocity if base_angular_velocity is not None else jnp.zeros(3)
+            (
+                base_angular_velocity
+                if base_angular_velocity is not None
+                else jnp.zeros(3)
+            ),
+            dtype=float,
         ).squeeze()
 
-        gravity = jnp.zeros(3).at[2].set(-standard_gravity)
+        gravity = jnp.zeros(3, dtype=float).at[2].set(-standard_gravity)
```

Comment:
The `dtype` of `jnp.zeros` is `float` by default, but we can leave it here for safety
```suggestion
        gravity = jnp.zeros(3).at[2].set(-standard_gravity)
```

Reply:
Can be removed, done in https://github.com/ami-iit/jaxsim/pull/256/commits/247826f66dee036f4779729092902408cf644ba1.

Reply:
@flferretti not sure if you realized that you blocked merging by asking changes. Given the nature of your comment, maybe it was a mistake :)